src/coverlet.core/Coverage.cs

```diff
@@ -80,23 +80,30 @@ public void PrepareModules()
                 }
 
                 var instrumenter = new Instrumenter(module, _identifier, _excludeFilters, _includeFilters, excludes, _excludeAttributes, _singleHit, _logger);
-                if (instrumenter.CanInstrument())
+                try
                 {
-                    InstrumentationHelper.BackupOriginalModule(module, _identifier);
-
-                    // Guard code path and restore if instrumentation fails.
-                    try
-                    {
-                        var result = instrumenter.Instrument();
-                        _results.Add(result);
-                        _logger.LogInformation($"Instrumented module: '{module}'");
-                    }
-                    catch (Exception ex)
+                    if (instrumenter.CanInstrument())
```

Comment:
Why not inside `CanInstrument()` prop?
We can log there and return false here, code seems more clean to me

Reply:
Ok, will change it tomorrow.

Reply:
Yup don't worry no rush!thank's for help!

Reply:
dome